File: backend/src/main/java/com/example/tbmini/domain/entity/Post.java
package com.example.tbmini.domain.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("posts")
public class Post {
    @TableId(type = IdType.AUTO)
    private Long id;

    private Long boardId;

    private Long userId;

    private String title;

    private String content;

    private String bodyMd;

    private String imageUrl;

    private String status;

    private Integer likeCount;

    private Integer commentCount;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;
}

File: frontend/src/api/post.ts
import request from './request'

export interface PostVo {
  id: number
  boardId: number
  boardName: string
  userId: number
  userNickname: string
  userAvatar: string
  title: string
  content: string
  imageUrl: string
  status: string
  likeCount: number
  liked: boolean
  createdAt: string
}

export interface PostDetailResponse {
  id: number
  boardId: number
  boardName: string
  userId: number
  userNickname: string
  userAvatar: string
  title: string
  bodyMd: string
  status: string
  likeCount: number
  liked: boolean
  createdAt: string
  updatedAt: string
  images: string[]
}

export interface PostListItemVo {
  id: number
  boardId: number
  boardName: string
  userId: number
  userNickname: string
  userAvatar: string
  title: string
  summary: string
  imageUrl: string
  likeCount: number
  commentCount: number
  createdAt: string
}

export function listPosts(params: { boardId?: number; page?: number; size?: number }) {
  return request.get('/posts', { params })
}

export function getPost(id: number) {
  return request.get<PostDetailResponse>(`/posts/${id}`)
}

export function createPost(data: FormData) {
  return request.post<PostDetailResponse>('/posts', data, {
    headers: { 'Content-Type': 'multipart/form-data' }
  })
}

export function likePost(id: number) {
  return request.post(`/posts/${id}/like`)
}

export function unlikePost(id: number) {
  return request.delete(`/posts/${id}/like`)
}

export function deletePost(id: number) {
  return request.delete(`/posts/${id}`)
}
